src/hooks/useAppState/reducer.js

```diff
@@ -53,37 +81,77 @@ const reducer = (state, { type, data }) => {
       const { key } = data;
       return {
         ...state,
-        items: state.items.map((item) => {
-          if (item.key !== key) {
-            return item;
+        items: state.items.map((items, index) => {
+          if (index !== state.index) {
+            return items;
           }
-          return {
-            ...item,
-            isCompleted: !item.isCompleted,
-          };
+          return items.map((item) => {
+            if (item.key !== key) {
+              return item;
+            }
+            return {
+              ...item,
+              isCompleted: !item.isCompleted,
+            };
+          });
         }),
       };
     }
 
     case types.STORE_ID: {
-      const items = state.items.map((item) => {
-        if (item.key !== data.key) {
-          return item;
-        }
-        return {
-          ...item,
-          id: data.id,
-        };
-      });
-      return { ...state, items };
+      const { key, id } = data;
+      return {
+        ...state,
+        items: state.items.map((items, index) => {
+          if (index !== state.index) {
+            return items;
+          }
+          return items.map((item) => {
+            if (item.key !== key) {
+              return item;
+            }
+            return {
+              ...item,
+              id,
+            };
+          });
+        }),
+      };
+    }
+
+    case types.STORE_LIST_ID: {
+      const { key, id } = data;
+      return {
+        ...state,
+        lists: state.lists.map((list) => {
+          if (list.key !== key) {
+            return list;
+          }
+          return {
+            ...list,
+            id,
+          };
+        }),
+      };
```

Comment:
We first create a list locally and then we call the api to create it, once we've done that we need to store the ID of the newly created list